GDJS/Runtime/howler-sound-manager/howler-sound-manager.ts

```diff
@@ -1,104 +1,296 @@
+    }
+
+    /**
+     * Get the sound seek.
+     */
+    getSeek(): float {
+      if (typeof this._id === 'undefined') return 0;
+      return this._howl.seek(this._id);
+    }
+
+    /**
+     * Set the sound seek.
+     * @returns The current instance for chaining.
+     */
+    setSeek(seek: float): this {
+      if (typeof this._id !== 'undefined') this._howl.seek(seek, this._id);
+      return this;
+    }
+
+    /**
+     * Get the sound spatial position.
+     */
+    getSpatialPosition(axis: 'x' | 'y' | 'z'): float {
+      if (typeof this._id === 'undefined') return 0;
+      return this._howl.pos(this._id)[axis === 'x' ? 0 : axis === 'y' ? 1 : 2];
+    }
+
+    /**
+     * Set the sound spatial position.
+     * @returns The current instance for chaining.
+     */
+    setSpatialPosition(x: float, y: float, z: float): this {
+      if (typeof this._id !== 'undefined') this._howl.pos(x, y, z, this._id);
+      return this;
     }
 
-    // Add methods to query the status of the sound:
-    paused() {
-      return this._paused;
+    /**
+     * Fade the volume sound.
+     * @returns The current instance for chaining.
+     */
+    fade(from: float, to: float, duration: float): this {
+      if (typeof this._id !== 'undefined')
+        this._howl.fade(from, to, duration, this._id);
+      return this;
     }
 
-    stopped() {
-      return this._stopped;
+    /**
+     * Adds an event listener to the howl.
+     */
+    on(event: HowlerEvent, handler: HowlCallback): this {
+      if (event === 'play') {
+        if (typeof this._id === 'undefined') this._onPlay.push(handler);
+        else this._howl.on(event, handler, this._id);
+      } else if (typeof this._id === 'undefined')
+        this.once('play', () => this.on(event, handler));
+      else this._howl.on(event, handler, this._id);
+
+      return this;
     }
 
-    canBeDestroyed() {
-      return this._canBeDestroyed;
+    /**
+     * Adds an event listener to the howl that removes itself after being called.
+     */
+    once(event: HowlerEvent, handler: HowlCallback): this {
+      if (event === 'play') {
+        if (typeof this._id === 'undefined') this._oncePlay.push(handler);
+        else this._howl.once(event, handler, this._id);
+      } else if (typeof this._id === 'undefined')
+        this.once('play', () => this.once(event, handler));
+      else this._howl.once(event, handler, this._id);
+
+      return this;
     }
 
-    // Methods to safely update the rate of the sound:
-    getRate() {
-      return this._rate;
+    /**
+     * Removes an event listener to the howl.
+     */
+    off(event: HowlerEvent, handler: HowlCallback): this {
+      if (this._id !== null) this._howl.off(event, handler, this._id);
+      return this;
     }
 
-    setRate(rate): void {
-      this._rate = HowlerSoundManager.clampRate(rate);
-      this.rate(this._rate);
+    /**
+     * Returns the raw howl.
+     * Be careful with this, you could break something!
```

Comment:
I would go as far as "Not to be used outside of the game engine." or mark this private and add a // @ts-ignore when using it.